saleor/graphql/meta/mutations.py

```diff
@@ -400,3 +402,98 @@ def perform_mutation(cls, _root, info, **data):
                 instance.delete_value_from_private_metadata(key)
             _save_instance(instance, "private_metadata")
         return cls.success_response(instance)
+
+
+class BaseMutationWithMetadata(BaseMutation):
```

Comment:
`BaseMetadataMutation` vs `BaseMutationWithMetadata` - sounds very similar, will be hard to distinguish. Don't have a better name yet but we could discuss.